### PR TITLE
Don't erase lines on :down if history disabled (#20).

### PR DESCRIPTION
**Pull request by cpence**
### Describe the change
As discussed in #20, don't erase the line on `:down` if `track_history?` is false.

### Requirements
Put an X between brackets on each line if you have done the item:
[X] Tests written & passing locally?
[X] Code style checked? (I don't know if you have a formal style, but I copied as best I could)
[X] Rebased with `master` branch?
[N/A] Documentaion updated?

### CODE REVIEW

File: lib/tty/reader.rb [hunk 1]
```diff
@@ -256,7 +256,7 @@ def read_line(prompt = "", **options)
         elsif console.keys[char] == :up
           line.replace(history_previous) if history_previous?
         elsif console.keys[char] == :down
-          line.replace(history_next? ? history_next : "")
+          line.replace(history_next? ? history_next : "") if track_history?
```
Comment (piotrmurach) [inline lib/tty/reader.rb:259]: How about replicating the `:up` behaviour?

```ruby
line.replace(history_next) if history_next?
```
Do you see any downsides?

**Review by piotrmurach — COMMENTED** [review]
Thanks! That was quick. Please see my comment. Great that you added tests for the case when history tracking is disabled. Would be good to capture also the case when history is enabled if it isn't already.

File: lib/tty/reader.rb (hunk 1 above)
Comment (cpence) [reply lib/tty/reader.rb:259]: I think you want that behavior when the history is enabled – for instance, in zsh, scrolling down "past the end" of the history does indeed give you a fresh blank command-line, which I think is a good idea.

**Review by piotrmurach — COMMENTED** [review]
As the last thing, could you please add an entry in the changelog as well? Then I'm happy to merge.

File: lib/tty/reader.rb (hunk 1 above)
Comment (piotrmurach) [reply lib/tty/reader.rb:259]: Ah, great point!

**Comment by cpence** [comment]
Added! I didn't squash the commits because I've never known the right git one-liner to do that...

**Comment by piotrmurach** [comment]
Thank you! ❤️

**Pull request merged**